src/clj/game/core/rezzing.clj

```diff
@@ -143,23 +143,50 @@
+  [state side eid cards {:keys [source-card and-then] :as msg-keys}]
+  (let [card-strs (enumerate-str (map #(card-str state % {:visible true}) cards))
+        prepend-cost-str (get-in msg-keys [:include-cost-from-eid :latest-payment-str])
+        title (or (:title source-card) (:printed-title source-card))]
+    (system-msg
+      state side
+      (cond
+        (not source-card) (str "derezzes " card-strs and-then)
+        prepend-cost-str (str prepend-cost-str " to use " title " to derez " card-strs and-then)
+        :else (str "uses " title " to derez " card-strs and-then)))))
+
 (defn derez
-  "Derez a corp card."
-  ([state side card] (derez state side card nil))
-  ([state side card {:keys [source-card no-msg] :as args}]
-   (let [card (get-card state card)]
-     (when-not no-msg
-       (system-msg state side (str (if source-card
-                                     (str "uses " (:title source-card) " to derez ")
-                                     "derezzes ")
-                                   (:title card))))
-     (unregister-events state side card)
-     (update! state :corp (deactivate state :corp card true))
-     (let [cdef (card-def card)]
-       (when-let [derez-effect (:derez-effect cdef)]
-         (resolve-ability state side derez-effect (get-card state card) nil))
-       (when-let [derezzed-events (:derezzed-events cdef)]
-         (register-events state side card (map #(assoc % :condition :derezzed) derezzed-events))))
-     (unregister-static-abilities state side card)
-     (update-disabled-cards state)
-     (trigger-event state side :derez {:card card :side side}))))
+  "Derez a number of corp cards."
+  ([state side eid cards] (derez state side eid cards nil))
+  ([state side eid cards {:keys [source-card suppress-checkpoint no-event no-msg msg-keys] :as args}]
+   (let [cards (if (sequential? cards)
+                 (filterv #(and (get-card state %) (rezzed? %)) (flatten cards))
+                 [cards])]
+     (if-not (seq cards)
+       (effect-completed state side eid)
+       (do (doseq [c cards]
+             (unregister-events state :corp c)
+             (update! state :corp (deactivate state :corp c true))
+             (let [cdef (card-def c)]
+               (when-let [derez-effect (:derez-effect cdef)]
+                 ;; this is currently only for lycian fixing subtypes on derez
+                 ;; should happen even if the card is disabled - nbk
+                 (resolve-ability state :corp derez-effect (get-card state c) cdef))
+               (when-let [derezzed-events (:derezzed-events cdef)]
+                 (register-events state :corp c (map #(assoc % :condition :derezzed) derezzed-events))))
+             (unregister-static-abilities state :corp c))
+           (update-disabled-cards state)
+           (when-not no-event
+             (queue-event state :derez {:cards cards
+                                        :side side}))
```

Comment:
in which cases would we not want to queue an event?

Reply:
no other handlers support this and none of your code in this PR uses it, so I don't think we should add it.

Reply:
oops, I was meant to flag that as true for the derez command and the derez button on the gameboard (these are for fixing mistakes, and shouldn't fire events)